src/codegen/object.rs

```diff
@@ -141,7 +141,10 @@ pub fn generate(
 
     if need_generate_inherent(analysis) {
         writeln!(w)?;
-        write!(w, "impl {} {{", analysis.name)?;
+        if analysis.should_generate_impl_block() {
```

Comment:
```suggestion
        if !analysis.should_generate_impl_block() {
            return Ok(());
        }
```

would be even easier or not? Or to otherwise skip the whole block below

Reply:
Ah Indeed we can simplify it a bit more, will update it in a bit